apps/rush-lib/src/logic/CredentialCache.ts

```diff
@@ -40,12 +40,7 @@ export interface ICredentialCacheOptions {
 /**
  * @beta
  */
-export type ICredentialCache = CredentialCache;
-
-/**
- * @beta
- */
-export class CredentialCache implements IDisposable {
+export class CredentialCache /* implements IDisposable */ {
```

Comment:
Why are you removing the `IDisposable`?

Reply:
If we want to make an `IDisposable` public API, it should probably go in `node-core-library`, and should go together with some kind of contract. For example maybe there is a helper API similar to the C# `using (new DisposableThing()) {`.  Or maybe `IDisposable` objects get registered with a manager that takes care of disposing them automatically when the session goes out of scope.

In its current form, `IDisposable` didn't seem to serve any useful purpose besides implying that whoever constructs `CredentialCache` has a responsibility to call `dispose()`.  But is that even the design we want?  For example, maybe it would be better for plugins to request `CredentialCache` from `RushSession` or a `BuildCacheProviderBase` class.  And then Rush itself can take care of disposing `CredentialCache`.  And maybe the `CredentialCache` constructor should be internal.

Reply:
Fair enough.